### DISCUSSION TITLE
VIC 4.1.2 version running error

### DISCUSSION THREAD

**Discussion by Chanhyuk-Choi** [NASA-LIS/LISF#1310] [Q&A]
Hello, 

I trying to VIC 4.1.2 version for LIS 7.4.0

I executed ldt.config file successfully, obtained output file (lis.nc file), and executed the lis.config file using it.

However, an error occurs when continuously reading the vegetation parameter file. This is as follows. I got the vegetation parameter file this link https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/VIC_PARAMETERS/1deg/

I don't understand how solve it. can you help with this issues?

I have attacthed about my ldt, lis config file and contents of the error.
 
[error.txt](https://github.com/NASA-LIS/LISF/files/11230529/error.txt)

[lis.config.era5.vic4.1.1.txt](https://github.com/NASA-LIS/LISF/files/11230534/lis.config.era5.vic4.1.1.txt)
[ldt.config_from_noah.txt](https://github.com/NASA-LIS/LISF/files/11230535/ldt.config_from_noah.txt)

**Comment by Chanhyuk-Choi** [comment]
Hello, @dmocko 

Can I ask about my error in running VIC412?

Thank you.

**Comment by dmocko** [comment]
Hello @Choi-chanhyuck 

VIC-4.1.2 in LIS only works with the AVHRR landcover classification.  This is both true in the code itself, as well as in the vegetation parameter file that you are using.  This is also given in the README file in the same directory as you listed above.

Change these lines in your ldt.config file to:

```
Landcover data source:                  "AVHRR"
Landcover classification:               "UMD"
Landcover file:                         ./data/landcover_UMD.1gd4r
```

You can obtain this global 1-km AVHRR data here:
    https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/UMD/1KM/landcover_UMD.1gd4r.gz

You also need to change your "Landmask data source:", but you can continue to "create" your mask.

**Comment by Chanhyuk-Choi** [comment]
Hello, @dmocko 

Thank you for your reply, 
I changed the Landcover and Landmask script in ldt.config file as you told me.

but after that I still got an error for reading related vic vegetation library and parameter file..
I think veg file has not vegetation type number up to 13 types but it seems to ask for the 13th type when reading..

Can you please check this error?


Thank you dmocko.

1. error when reading vic vegetation parameter files
![image](https://user-images.githubusercontent.com/73869049/234452469-82e848a0-49fa-4f9d-8781-0177e9db3e6c.png)

2. vic ldt.config file
[ldt.config.vic411_1.0grid_era5.txt](https://github.com/NASA-LIS/LISF/files/11328427/ldt.config.vic411_1.0grid_era5.txt)

3. lis.config file
[lis.config.era5.vic4.1.1.txt](https://github.com/NASA-LIS/LISF/files/11328428/lis.config.era5.vic4.1.1.txt)

**Reply by dmocko** [reply]
Hmmmm

Perhaps "readin" the landmask isn't correct here.  You may need to use this in your ldt.config:

```
Create or readin landmask:              "create"
Landmask data source:                   AVHRR
```
When doing "create", you can remove all other config entries for "Landmask".

**Reply by Chanhyuk-Choi** [reply]
Okay, @dmocko

I will proceed with your advice.
I'm also having errors with Mosaic, can you give me some advice?
#991

Thank you for your kind reply.

Choi.

**Reply by Chanhyuk-Choi** [reply]
Hi, @dmocko 

As you advised, I changed the "readin" setting to "create" of Landmask setting and ran it again, and the same error is occurring in reading vegetation parameter file..

Can I get some help to solve it?

Thank you.

**Comment by dmocko** [comment]
Sorry - I am trying to help, but I haven't run VIC in a long time, and never using these datasets.

Try using the landcover and global_mask files that are available on the portal:

https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/VIC_PARAMETERS/1deg/

Still use AVHRR and UMD for your landcover data source and classification.

Change your ldt.config file again to make your landcover and landmask dimensions:

lower left lat: -59.5
lower left lon: -179.5
upper right lat: 89.5
upper right lon: 179.5
dx/dy dimensions: 1.0

You can still have a different "Run domain" that's only part of this total domain, so you don't need to change that.

Then, change back "readin" landmask, and point to the landmask file.

Take a look at your lis_input file when done.  If the values don't seem realistic, you may need to change what you chose for "endian" when you configured LDT.

**Reply by Chanhyuk-Choi** [reply]
Hi @dmocko

As advised, I tried changing the ldt settings and running it,
I'm getting an error that I can't understand.

It's probably an error when reading global_mask,

Did I set it up wrong?

These are the ldt.config file and the error file.

thank you.

[ldt.config.vic411_1.0grid_era5.txt](https://github.com/NASA-LIS/LISF/files/11345599/ldt.config.vic411_1.0grid_era5.txt)

[error.txt.txt](https://github.com/NASA-LIS/LISF/files/11345601/error.txt.txt)

**Reply by hkato1** [reply]
Hi @Choi-chanhyuck 

David forwarded me your questions on running VIC 4.1.2 globally.
I ran GLDAS VIC with slightly different parameter datasets, but I was able to run with the parameters on the portal and similar configuration as yours.  I believe the issue is that the landcover and landmask data files specified in the ldt.config was not correct.
Please see attached config file and output file from LDT.  Also, I made a test run with the lis.config using the lis_input file created.    
There are minor differences from your setting (i.e. forcing data and elevation) but they should be easily swapped.  
Hope this helps. Please let me know if you have any questions.

Best regards,
Hiroko
[ldt.config.vic412.org.txt](https://github.com/NASA-LIS/LISF/files/11375731/ldt.config.vic412.org.txt)
[lis.config_vic412.org.txt](https://github.com/NASA-LIS/LISF/files/11375732/lis.config_vic412.org.txt)
[vic412param_org.zip](https://github.com/NASA-LIS/LISF/files/11375741/vic412param_org.zip)

**Reply by Chanhyuk-Choi** [reply]
Hi @hkato1 

Thank you for your kind reply,

I ran it right away as you told me, I changed the land cover and land mask settings a little and tried again.

but this time I'm getting a new error In the OUTPUT LIST file, set all variables to 0.

Could it be a GLDAS forcing problem?

Thank you,
Chanhyuk


[ldt.config.vic412_1.0grid_era5.txt](https://github.com/NASA-LIS/LISF/files/11380210/ldt.config.vic412_1.0grid_era5.txt)
[lis.config.era5.vic4.1.2.txt](https://github.com/NASA-LIS/LISF/files/11380211/lis.config.era5.vic4.1.2.txt)
[error.txt](https://github.com/NASA-LIS/LISF/files/11380212/error.txt)
[lislog.00.0000.txt](https://github.com/NASA-LIS/LISF/files/11380213/lislog.00.0000.txt)

**Reply by hkato1** [reply]
Hi @Choi-chanhyuck 

Based on the log file, it seems that the forcing fields are read in without problem, but it crashes when writing out the first variable Swnet.  Please compare your MODEL_OUTPUT_LIST.TBL against the one attached here and double check.

I noticed that some of the options for the VIC model in the lis.config file are different from mine as well.  Try changing as below (the text file may have *.txt extension):
VIC412 total number of veg types:  13
VIC412_LAI_SRC:                     LAI_FROM_VEGPARAM
VIC412_SNOW_BAND:                   25 ./input/VIC_PARAMETERS/1deg/snowband_1.0_25

Not sure if these matter but I have the following:
Use greenness fraction climatology: 0
Use albedo climatology: 0

I forgot to mention that VIC prints out a bunch of warnings, but you can safely ignore them.

Best regards,
Hiroko
[MODEL_OUTPUT_LIST.TBL.test.txt](https://github.com/NASA-LIS/LISF/files/11383488/MODEL_OUTPUT_LIST.TBL.test.txt)

**Reply by Chanhyuk-Choi** [reply]
Hello, @hkato1 

Sorry,

I'm getting the same error even though I set all the settings as written above.
I really want to solve it, but I'm not sour how.. It's a pity

[lis.config.era5.vic4.1.2.txt](https://github.com/NASA-LIS/LISF/files/11398783/lis.config.era5.vic4.1.2.txt)
[error.txt](https://github.com/NASA-LIS/LISF/files/11398784/error.txt)
[MODEL_OUTPUT_LIST_VIC.TBL.txt](https://github.com/NASA-LIS/LISF/files/11398785/MODEL_OUTPUT_LIST_VIC.TBL.txt)

**Reply by hkato1** [reply]
Hi @Choi-chanhyuck 

OK, so maybe the problem is related to the forcing.  I see the separate email thread below about the GLDAS forcing, but it is your own code?
I believe there's GLDAS met forcing as one of the testcases (see testcases/metforcing/gldas).  Do you have access to the testcase data?  
I can check that GLDAS forcing testcase works with VIC and get back to you.
Assuming the lis_input file you've created is working, we should use a testcase meteorological forcing to make sure that VIC runs on your end. Then you can move onto debugging your code.  Does it sound like a plan?

Best regards,
Hiroko

**Comment by dmocko** [comment]
Looks like the landcover/landmask data is only 360 X 140 points.

Try making your upper right lat for both only 79.5 instead of 89.5.

Meanwhile, I will look further into this, and ask for help from the team.

**Reply by Chanhyuk-Choi** [reply]
I tried following your advice, but got an same error

When I tried with the same settings as Run domain, the ldt file was created.

The output nc file is as follows, and the landmask fillvalue seems to be incorrect.
Please check once.

[ldt.config.vic411_1.0grid_era5 (2).txt](https://github.com/NASA-LIS/LISF/files/11345862/ldt.config.vic411_1.0grid_era5.2.txt)
[lis_input.gldas_1.0grid_vic4.1.2.d01.zip](https://github.com/NASA-LIS/LISF/files/11345870/lis_input.gldas_1.0grid_vic4.1.2.d01.zip)

**Reply by Chanhyuk-Choi** [reply]
Hello, @dmocko 

The parameter reading error mentioned above when driving the VIC model seems to have been solved.

However, I am getting a GLDAS forcing reading error, and I don't know why this error is occurring. 

Can you please confirm this error?

[lislog.00.0000.txt](https://github.com/NASA-LIS/LISF/files/11350938/lislog.00.0000.txt)
[lis.config.era5.vic4.1.1.txt](https://github.com/NASA-LIS/LISF/files/11350939/lis.config.era5.vic4.1.1.txt)

**Reply by dmocko** [reply]
I don't have your GLDAS forcing reader code, so I can't diagnose what is going on.  You'll have to figure this one yourself, or share the code.

Once it is working, we would be interested in potentially having you open a pull request, so we can add this new capability into our code.

**Reply by Chanhyuk-Choi** [reply]
Thank you for your reply, @dmocko 
 
I will share my GLDAS metforcing code in LISF_7.4.0/metforcing/gldas to you 
However, since I received and used the code modified by someone else, I do not understand it correctly.

If there is another way, can you recommend it?
I want to receive ERA5 data in real time and prescribe it to the ground model in the form of an ncfile.

However, I tried it with the ERA5 setting, but I couldn't solve it because I was experiencing a G2P reading error.

[gldas.zip](https://github.com/NASA-LIS/LISF/files/11354747/gldas.zip)
